src/dsp_tools/utils/date_util.py

```diff
@@ -8,16 +8,21 @@
 
 from dsp_tools.models.exceptions import BaseError
 
-_full_date_pattern = r"""
+_calendar = r"GREGORIAN|JULIAN|ISLAMIC"
+_era = r"CE|BCE|BC|AD"
+_year = r"\d{1,4}"
```

Comment:
here we would need to specify r"[1-9]\d{0,3}"

Reply:
There might well be users who pad a date with zeroes. And even our test data often makes it, see here: https://github.com/dasch-swiss/dsp-tools/blob/4afc63ff47f117f6632de5b4316b628942c9e396/testdata/xml-data/test-data-systematic.xml#L117. 
So I don't think it would be wise to do that.

And besides, I don't think it would solve the validation problem: If we allow 1-3 digit years, it is impossible (or at least very tricky) to forbid the test case I have removed.

Reply:
Ok. I understand your case.